src/cljs/athens/views/node_page.cljs

```diff
@@ -179,6 +160,97 @@
+  TODO: When user presses enter, blur
+    if there are block refs, update them
+    if existing node/title, prompt to merge"
+  [e _state]
+  (let [{:keys [key-code shift]} (destruct-event e)]
+    (when (and (not shift) (= key-code KeyCodes.ENTER))
+      (.. e preventDefault))))
+
+
+(defn handle-change
+  [e state]
+  (let [value (.. e -target -value)]
+    (swap! state assoc :current-title value)))
+
+
+(defn get-linked-refs
+  [ref-groups]
+  (->> ref-groups
+       first
+       second
+       first
+       second))
+
+
+(defn map-new-refs
+  [linked-refs old-title new-title]
+  (map (fn [{:block/keys [uid string order]}]
+         (let [new-str (str/replace string
+                                    (patterns/linked old-title)
+                                    (str "$1$3$4" new-title "$2$5"))]
+           {:db/id [:block/uid uid]
+            :block/string new-str
+            :block/order order}))
+       linked-refs))
+
+
+(defn handle-blur
+  "When textarea blurs and its value is different from initial page title:
+   - if no other page exists, rewrite page title and all linked refs
+   - else page with same title does exists: prompt to merge
+     - confirm-fn: rewrite page title, linked refs, AND merge
+     - cancel-fn: reset state"
+  [block state ref-groups]
+  (let [{dbid :db/id children :block/children} block
+        {:keys [old-title current-title]} @state]
+    (when (not= old-title current-title)
+      (let [existing-page (d/q '[:find ?e ?uid ?b
+                                 :in $ ?t
+                                 :where
+                                 [?e :node/title ?t]
+                                 [?e :block/uid ?uid]
+                                 [?e :block/children ?b]]
+                               @db/dsdb current-title)
+            linked-refs (get-linked-refs ref-groups)
+            new-linked-refs (map-new-refs linked-refs old-title current-title)]
+        (if (empty? existing-page)
+          (let [new-page {:db/id dbid :node/title current-title}]
+            (swap! state assoc :old-title current-title)
+            (dispatch [:transact (concat [new-page] new-linked-refs)]))
+          (let [new-parent (ffirst existing-page)
+                new-parent-uid (-> existing-page first second)
+                existing-page-block-count (count existing-page)
+                reindex (map (fn [{:block/keys [order uid]}]
+                               {:db/id [:block/uid uid]
+                                :block/order (+ order existing-page-block-count)
+                                :block/_children new-parent})
+                             children)
+                delete-page [:db/retractEntity dbid]
+                new-datoms (concat [delete-page]
+                                   new-linked-refs
+                                   reindex)
+                cancel-fn #(swap! state assoc
+                                  :current-title old-title
```

Comment:
Is it better to have all these key-value pairs, or to create a new map that is `merge`d? The latter gives you a nice data structure, though I don't see how or if that would be used yet.